edge/cmd/edgecore/app/server.go

```diff
@@ -82,12 +85,19 @@ offering HTTP client capabilities to components of cloud to reach HTTP servers r
 				}
 			}
 
-			// get edge node local ip
-			if config.Modules.Edged.NodeIP == "" {
-				hostnameOverride := util.GetHostname()
-				localIP, _ := util.GetLocalIP(hostnameOverride)
+			// Get edge node local ip only when the custiomInterfaceName has been set.
+			// Defaults to the local IP from the default interface by the default config
+			if config.Modules.Edged.CustomInterfaceName != constants.DefaultCustomInterfaceName {
+				var ip net.IP
+				var localIP string
+				klog.Infof("Getting IP address by custom interface: %s", config.Modules.Edged.CustomInterfaceName)
+				ip, err = netutil.ChooseBindAddressForInterface(config.Modules.Edged.CustomInterfaceName)
+				if ip != nil {
+					localIP = ip.String()
+				}
```

Comment:
If `err != nil` or `ip == nil` , do we need stop the process?

Reply:
+1, please handle the err case.

Reply:
Ok. will exit the process in the error case.